modules/core/src/main/scala/ch/epfl/bluebrain/nexus/admin/core/resources/Resources.scala

```diff
@@ -45,28 +47,36 @@ abstract class Resources[F[_], A: IdResolvable: PersistenceId: TypeFilterExpr](a
                                                                                sparqlClient: SparqlClient[F])(
     implicit
     F: MonadError[F, Throwable],
+    validator: ShaclValidator[F],
     logger: Logger,
     clock: Clock) {
 
-  /**
-    * Certain validation to take place during creation operations.
-    *
-    * @param id    the identifier of the resource
-    * @param value the json payload of the resource
-    * @return () or the appropriate rejection in the ''F'' context
-    */
-  @silent
-  def validateCreate(id: A, value: Json): F[Unit]
+  def resourceType: IdRef
+
+  def resourceSchema: Json
 
   /**
-    * Certain validation to take place during update operations.
+    * Certain validation to perform on the payload.
     *
     * @param id    the identifier of the resource
     * @param value the json payload of the resource
     * @return () or the appropriate rejection in the ''F'' context
     */
-  @silent
-  def validateUpdate(id: A, value: Json): F[Unit]
+  def validate(id: A, value: JsonLD): F[Unit] = {
+    value.appendContext(resourceContext).add(rdf.tpe, resourceType).add(nxv.label, id.value).apply() match {
```

Comment:
I thought the label was going to be the `uuid`. But I'm not sure that's clear yet, so we can change that later